frontend/src/screens/internal-apps/ZapPlanner.tsx

```diff
@@ -0,0 +1,183 @@
+  CardHeader,
+  CardTitle,
+} from "src/components/ui/card";
+import { Input } from "src/components/ui/input";
+import { LoadingButton } from "src/components/ui/loading-button";
+import { useToast } from "src/components/ui/use-toast";
+import { useApps } from "src/hooks/useApps";
+import { createApp } from "src/requests/createApp";
+import { CreateAppRequest } from "src/types";
+import { handleRequestError } from "src/utils/handleRequestError";
+
+type Recipient = {
+  name: string;
+  description: string;
+  lightningAddress: string;
+};
+
+const recipients: Recipient[] = [
+  {
+    name: "Alby",
+    description:
+      "Support the open source development of Alby Hub, Alby Go, Alby developer tools, NWC protocol development, and more.",
+    lightningAddress: "hello@getalby.com",
+  },
+  {
+    name: "HRF",
+    description:
+      "We collaborate with transformative activists to develop innovative solutions that bring the world together in the fight against tyranny.",
+    lightningAddress: "hrf@btcpay.hrf.org",
+  },
+  {
+    name: "OpenSats",
+    description:
+      "Help us to provide sustainable funding for free and open-source contributors working on freedom tech and projects that help bitcoin flourish.",
+    lightningAddress: "opensats@btcpay0.voltageapp.io",
+  },
+];
+
+export function ZapPlanner() {
+  const { data: apps, mutate: reloadApps } = useApps();
+  const { toast } = useToast();
+  const [submittingFor, setSubmittingFor] = React.useState("");
+  const [customName, setCustomName] = React.useState("");
+  const [customLightningAddress, setCustomLightningAddress] =
+    React.useState("");
+  const handleSubmit = async (name: string, lightningAddress: string) => {
+    setSubmittingFor(lightningAddress);
+
+    try {
+      if (apps?.some((existingApp) => existingApp.name === name)) {
+        throw new Error("A connection with the same name already exists.");
+      }
+
+      const amountSats = 5000;
+
+      const createAppRequest: CreateAppRequest = {
+        name: `ZapPlanner - ${name}`,
+        scopes: [
+          "get_balance",
+          "get_info",
+          "list_transactions",
+          "lookup_invoice",
+          "make_invoice",
+          "notifications",
+          "pay_invoice",
+        ],
+        budgetRenewal: "monthly",
+        maxAmount: Math.floor(amountSats * 1.01) + 10 /* with fee reserve */,
+        isolated: false,
+        metadata: {
+          app_store_app_id: "zapplanner",
+          recipient_lightning_address: lightningAddress,
+        },
+      };
+
+      const createAppResponse = await createApp(createAppRequest);
+      /*toast({
+        title: "Created subscription",
+        description: "The first payment is scheduled for 1st of October.",
+      });*/
+      const comment = encodeURIComponent("ZapPlanner payment from Alby Hub"); // TODO: allow customization
+      const payerData = ""; //encodeURIComponent(JSON.stringify({}));
+      // TODO: consider fiat conversion
```

Comment:
remove this TODO - we also need fiat budgets which are unlikely to be added soon